avocado/utils/nfs.py

```diff
@@ -0,0 +1,349 @@
+            path.find_command("exportfs")
+            self.nfs_service = service.SpecificServiceManager("nfs")
+
+            self.export_dir = (params.get("export_dir") or
+                               self.mount_src.split(":")[-1])
+            self.export_ip = params.get("export_ip", "*")
+            self.export_options = params.get("export_options", "").strip()
+            self.exportfs = Exportfs(self.export_dir, self.export_ip,
+                                     self.export_options)
+            self.mount_src = "127.0.0.1:%s" % self.export_dir
+
+    def is_mounted(self):
+        """
+        Check the NFS is mounted or not.
+
+        :return: If the src is mounted as expect
+        :rtype: Boolean
+        """
+        return utils_misc.is_mounted(self.mount_src, self.mount_dir, "nfs")
+
+    def mount(self):
+        """
+        Mount source into given mount point.
+        """
+        return utils_misc.mount(self.mount_src, self.mount_dir, "nfs",
+                                perm=self.mount_options)
+
+    def umount(self):
+        """
+        Umount the given mount point.
+        """
+        return utils_misc.umount(self.mount_src, self.mount_dir, "nfs")
+
+    def setup(self):
+        """
+        Setup NFS in host.
+
+        Mount NFS as configured. If a local nfs is requested, setup the NFS
+        service and exportfs too.
+        """
+        if self.nfs_setup:
+            if not self.nfs_service.status():
+                logging.debug("Restart NFS service.")
+                self.nfs_service.restart()
+
+            if not os.path.isdir(self.export_dir):
+                os.makedirs(self.export_dir)
+            self.exportfs.export()
+            self.unexportfs_in_clean = not self.exportfs.already_exported
+
+        logging.debug("Mount %s to %s" % (self.mount_src, self.mount_dir))
+        if os.path.exists(self.mount_dir):
+            if not os.path.isdir(self.mount_dir):
+                raise OSError(
+                    "Mount point %s is not a directory, check your setup." %
+                    self.mount_dir)
+
+        if not os.path.isdir(self.mount_dir):
+            os.makedirs(self.mount_dir)
+            self.mk_mount_dir = True
+        self.mount()
+
+    def cleanup(self):
+        """
+        Clean up the host env.
+
+        Umount NFS from the mount point. If there has some change for exported
+        file system in host when setup, also clean up that.
+        """
+        self.umount()
+        if self.nfs_setup and self.unexportfs_in_clean:
+            self.exportfs.reset_export()
+        if self.mk_mount_dir and os.path.isdir(self.mount_dir):
+            utils_misc.safe_rmdir(self.mount_dir)
+
+
+class NFSClient(object):
+
+    """
+    NFSClient class for handle nfs remotely mount and umount.
+    """
+
+    def __init__(self, params):
+        # Setup SSH connection
+        print params
+        self.ssh_obj = SSHConnection(params)
```

Comment:
This is another unwanted dependency. Take a look at `avocado.utils.service.get_name_of_init`. You can simply expect a `process.run`-like object which can be either local, or remote and use it to execute commands. This would be definitely preferred as it's a standard, well behaved way and allows both, local and remote execution.

PS: I noticed that `remoter` is actually not available in `utils` so the users would have to import it from `avocado.core`. I created a card: https://trello.com/c/nzeIh2ky/897-move-avocado-core-remoter-to-avocado-utils-remoter to move it, because it's suppose to be generic interface like `avocado.utils.process` is.